### DISCUSSION TITLE
Dont know how to install .NET on raspberry pi

### DISCUSSION THREAD

**Discussion by feeshmaster** [dotnet/runtime#97051] [Q&A]
So when I tried to do it using many different tutorials, I couldnt figure it out, could someone explain how to get it on my raspberry pi and explain it in a way that a beginner that doesn't really know Linux could understand step by step

**Comment by jeffschwMSFT** [comment]
One of the likely easiest ways is to deploy .NET Core with your app.  Here is an article on building self-contained.

https://learn.microsoft.com/en-us/dotnet/core/deploying/#publish-self-contained

Eg. 
```
dotnet publish -c Release -r linux-arm64
```

**Comment by wfurt** [comment]
It probably should be `linux-arm` - Raspbian is 32 OS by default. You can check `cat /proc/cpuinfo` or `file /bin/bash`
`ARMv7` or ` ELF 32-bit LSB executable, ARM` is sign of 32 bit. 

Raspbian is based on Debian but the general instructions will not work since we don't seems to publish arm32 packages. You can fetch binaries from:
https://github.com/dotnet/installer/ - pick the `Linux arm` (not arm64)

**Comment by MagicAndre1981** [comment]
> It probably should be `linux-arm` - Raspbian is 32 OS by default. 

since [3B the recommended option is now 64 bit](https://www.raspberrypi.com/software/operating-systems/#raspberry-pi-os-64-bit) in the imager. 

![image](https://github.com/dotnet/runtime/assets/8274816/6addf1f8-db2a-4e46-a6fe-b31bf8655359)

He needs to check which version he uses.
